backend/app/core/events.py

```
@@ -164,6 +182,65 @@ async def publish(self, event: T) -> None:
+        try:
+            asyncio.run_coroutine_threadsafe(run_handlers(), self._main_loop)
+            logger.debug(
+                "[EVENT_BUS] Scheduled %d handlers in main loop for event %s",
+                len(handlers),
+                type(event).__name__,
+            )
+        except Exception as e:
+            logger.exception(
+                "[EVENT_BUS] Failed to schedule handlers in main loop: %s, "
+                "executing in current context",
+                e,
```

Ruff (0.14.14)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)